includes/plugins/wc-memberships/class-memberships.php

```diff
@@ -72,6 +75,10 @@ public static function init() {
 		add_filter( 'newspack_gate_content', 'wp_replace_insecure_home_url' );
 		add_filter( 'newspack_gate_content', 'do_shortcode', 11 ); // AFTER wpautop().
 
+		/** Fixes to ensure that memberships are linked to the correct active subscription. */
+		add_action( 'woocommerce_subscription_status_updated', [ __CLASS__, 'check_user_memberships_on_subscription_update' ] );
```

Comment:
```suggestion
		add_action( 'woocommerce_subscription_status_updated', [ __CLASS__, 'check_user_memberships_on_subscription_update' ], 11 );
```

This should run after the Memberships hook, otherwise resetting the end date may not stick